src/runtime-rs/crates/hypervisor/src/qemu/inner.rs

```diff
@@ -149,6 +152,14 @@ impl QemuInner {
             tokio::spawn(log_qemu_stderr(qemu_process.stderr.take().unwrap()));
         }
 
+        match Qmp::new(QMP_SOCKET_FILE) {
```

Comment:
This can legitimately fail because you have no guarantee that QEMU has called `listen()` yet.

Two options to address that :

1. pre-connect as the go runtime does (see https://github.com/kata-containers/kata-containers/blob/main/src/runtime/virtcontainers/qemu.go#L964)
2. loop for some time until connection succeeds

Reply:
I think just 2 is an option then since we don't do the connecting ourselves (qapi-rs does).

Reply:
AFAICT qapi-rs does the QMP handshake but the connection is done by qemu-rs at https://github.com/kata-containers/kata-containers/pull/9772/commits/e26028c34cdcde085b866cc7d764284c3c46824a#diff-d564620b2876afa7b1db8ade06afda6d28c187cb887d27880af3fd46866d1362R16. If this gets moved out to the caller, it is possible to implement 1.

Reply:
Right, I might be misunderstanding then what exactly it takes to implement the pre-connect option.

Reply:
As it turns out this PR already does the pre-connect like the go runtime linked above.  The go code basically 1) `Listen()`s on the qmp socket path, 2) duplicates that socket, 3) `Dial()`s the qmp socket path, and 4) returns the connection from 3, stores the duplicate from 2 and lets the original from 1 die.

To see that we're doing all that already, please note that `UnixListener::bind()` (executed in `QmpSocket::new()`) does include a `listen()` call so we have an equivalent of 1.  Then 2 is not essential for the desired functionality and is apparently just a side effect of golang's `UnixListener::File()` implementation, we do 3 in this PR in `Qmp::new()` and we do our equivalents of 4, too.

So while I'll readily admit that this just by pure luck, I submit that this PR complies already. :-)